test/unit/module_utils/test_device.py

```diff
@@ -44,6 +49,16 @@ def test_supports_ftd_model_should_return_false_for_non_supported_models(self):
         assert not Ftd2100Platform.supports_ftd_model(FtdModel.FTD_ASA5508_X.value)
         assert not FtdAsa5500xPlatform.supports_ftd_model(FtdModel.FTD_2120.value)
 
+    def test_parse_rommon_file_location(self):
+        server, path = AbstractFtdPlatform.parse_rommon_file_location('tftp://1.2.3.4/boot/rommon-boot.foo')
+        assert '1.2.3.4' == server
+        assert '/boot/rommon-boot.foo' == path
+
+    def test_parse_rommon_file_location_should_fail_for_non_tftp_protocol(self):
+        with pytest.raises(ValueError) as ex:
+            AbstractFtdPlatform.parse_rommon_file_location('http://1.2.3.4/boot/rommon-boot.foo')
+            assert 'The ROMMON image must be downloaded from TFTP server' in str(ex)
```

Comment:
shouldn't we use assert outside of with block?

Reply:
Good point, moved.